### ISSUE TITLE
python-dateutil dependency issue on install

### ISSUE THREAD

**Issue by challs** [home-assistant/appdaemon#780]
Hello, thank you for providing appdaemon.

I thought I'd try setting up a new venv from the dev branch. Unfortunately its broken at the moment:

```
% git clone -b dev https://github.com/home-assistant/appdaemon.git                                                                                      Cloning into 'appdaemon'...
% cd appdaemon                                                                                                                                          

~/dev/appdaemon-dateutil/appdaemon • python3 -m venv env                                                                                                                                     ~/dev/appdaemon-dateutil/appdaemon • source env/bin/activate                                                                                                                                 ~/dev/appdaemon-dateutil/appdaemon • pip install -e .                                                                                                                                        Obtaining file:///home/chris/dev/appdaemon-dateutil/appdaemon
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/chris/dev/appdaemon-dateutil/appdaemon/setup.py", line 6, in <module>
        from appdaemon.utils import (__version__)
      File "/home/chris/dev/appdaemon-dateutil/appdaemon/appdaemon/utils.py", line 13, in <module>
        import dateutil.parser
    ImportError: No module named 'dateutil'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /home/chris/dev/appdaemon-dateutil/appdaemon/

```
The problem is due to `setup.py` importing `utils.py` in order to get at `__ version __`. The new version adds a load of other utility functions, one of which needs `python-dateutil`.

The problem can be worked around by moving the import in question into the function that actually uses it:

```
diff --git a/appdaemon/utils.py b/appdaemon/utils.py
index 83fc8be..28c52c8 100644
--- a/appdaemon/utils.py
+++ b/appdaemon/utils.py
@@ -10,7 +10,6 @@ import pstats
 import shelve
 import threading
 import datetime
-import dateutil.parser
 import copy
 import json
 from functools import wraps
@@ -446,6 +445,7 @@ def check_path(type, logger, inpath, pathtype="directory", permissions=None):
 
 
 def str_to_dt(time):
+    import dateutil.parser
     return dateutil.parser.parse(time)
 
```
 
I think, though, that it would be safer to move that version string out of utils.py to a place that can be imported alone by `setup.py` without needing to import the rest of `utils.py`.

It would probably make sense to adjust the users of __version__ to point to a new place, meaning the patch would be bigger. I didn't want to start on doing that without any feedback from yourselves.

Another solution would be to place the version in a plaintext file that could be read by `setup.py` and `utils.py` in the same way as the Python packaging guide recommends reading README.md in https://packaging.python.org/guides/making-a-pypi-friendly-readme/

```
this_directory = path.abspath(path.dirname(__file__))
with open(path.join(this_directory, 'README.md'), encoding='utf-8') as f:
    long_description = f.read()
```

**Comment by acockburn** [comment]
Thanks for flagging this, we'll get a fix into 4.0.0. I will probably just pull __version__ into it's own .py file, but I appreciate all the other suggestions P)

**Comment by basnijholt** [comment]
I posted another suggestion in https://github.com/home-assistant/appdaemon/issues/817.

**Comment by acockburn** [comment]
Also referenced in #817 with suggested solution

**Comment by acockburn** [comment]
This is fixed in the pending 4.0.0 release (I used @basnijholt  implementation)
